src/pyconverter/xml2py/writer.py

```diff
@@ -295,90 +391,246 @@ def write_source(
 
-    cmd_map : dict
-        Dictionary with this format: ``{"command_name": "python_name"}``.
-
-    path : str
+    package_path : str
         Path to the new package folder.
 
+    module_name_list : list
+        List of module names created.
+
+    package_structure :
+        Dictionary with the following format:
+        ``{'python_module_name': [{'python_class_name': python_names_list}]}``.
+
     Returns
     -------
     str
         Path to the new document page.
 
     """
+    library_name = get_config_data_value(config_path, "library_name_structured")
+    if library_name[0] == "src":
+        library_name.pop(0)
+    library_name = ".".join(library_name)
 
     doc_package_path = os.path.join(package_path, "doc/source")
     if not os.path.isdir(doc_package_path):
         os.makedirs(doc_package_path)
 
     doc_src = os.path.join(doc_package_path, "docs.rst")
     with open(doc_src, "w") as fid:
-        fid.write("Autosummary\n")
-        fid.write("===========\n\n")
-
-        fid.write(".. currentmodule:: pyconverter.generatedcommands\n\n")
-        fid.write(".. autosummary::\n")
-        fid.write("   :template: base.rst\n")
-        fid.write("   :toctree: _autosummary/\n\n")
-        for ans_name in commands:
-            if ans_name not in SKIP_XML:
-                cmd_name = cmd_map[ans_name]
-                fid.write(f"   {cmd_name}\n")
+        fid.write(
+            """
+API documentation
+==================
+
+.. toctree::
+   :maxdepth: 2
+   :hidden:\n\n
+"""
+        )
+        for module_name in package_structure.keys():
+            fid.write(f"   {module_name}/index.rst\n")
+    fid.close()
+
+    if package_structure is not None:
+        for module_folder_name, class_map in tqdm(
+            package_structure.items(), desc="Writing docs..."
+        ):
+            module_title = module_folder_name.replace("_", " ").capitalize()
+            module_folder = os.path.join(doc_package_path, f"{module_folder_name}")
+            if not os.path.isdir(module_folder):
+                os.makedirs(module_folder)
+            module_file = os.path.join(module_folder, f"index.rst")
+            with open(module_file, "w") as fid:
+                fid.write(f".. _ref_{module_folder_name}:\n\n")
+                fid.write(f"{module_title}\n")
+                fid.write("=" * len(module_title) + "\n\n")
+                fid.write(f".. toctree::\n")
+                fid.write(f"   :maxdepth: 2\n")
+                fid.write(f"   :hidden:\n\n")
+                for class_file_name in class_map.keys():
+                    fid.write(f"   {class_file_name}\n")
+            fid.close()
+            for class_file_name, class_structure in class_map.items():
+                class_name = class_structure[0]
+                method_list = class_structure[1]
+                class_file = os.path.join(module_folder, f"{class_file_name}.rst")
+                with open(class_file, "w") as fid:
```

Comment:
I think this should work...

```suggestion
            for class_file_name, (class_name, method_list)  in class_map.items():
                class_file = os.path.join(module_folder, f"{class_file_name}.rst")
                with open(class_file, "w") as fid:
```